lld/COFF/LTO.cpp

```diff
@@ -116,7 +117,18 @@ BitcodeCompiler::BitcodeCompiler(COFFLinkerContext &c) : ctx(c) {
 
   // Initialize ltoObj.
   lto::ThinBackend backend;
-  if (ctx.config.thinLTOIndexOnly) {
+  if (!ctx.config.DTLTODistributor.empty()) {
+    StringRef version = getenv("LLD_VERSION"); // For testing only.
+    if (version.empty())
+      version = ctx.saver.save(getLLDVersion());
+    backend = lto::createOutOfProcessThinBackend(
```

Comment:
When I read this code at first I was confused why it needed to have the arguments for the parallelization, I looked at the comment in LTO.h later and realized that the arguments where not used for the actual codegen, but for generating the index. But I wonder if we even need to pass that as an argument then - it reads to me like it's controlling the external process in some way.

Reply:
Thanks. It may be unnecessary. https://github.com/llvm/llvm-project/commit/2edd897a4227e481af33e8e43090ab088cd9d953 implies that the summary index and import list generation can be expensive though which is why this setting is provided. Also some distribution systems have a mechanism to limit how many processes they launch concurrently on the local machine - we could pass the parallelization level to the distributors (by adding it to the JSON) so that this setting may be applied by the distributor?